include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -843,18 +903,34 @@ transform(_ExecutionPolicy&& __exec, _Range1&& __rng, _Range2&& __result, _Unary
     const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec, __rng, __result);
 
     oneapi::dpl::__internal::__ranges::__pattern_walk_n(
-        __dispatch_tag, ::std::forward<_ExecutionPolicy>(__exec), [__op](auto x, auto& z) { z = __op(x); },
+        __dispatch_tag, ::std::forward<_ExecutionPolicy>(__exec), __internal::__transform_fn<_UnaryOperation>{__op},
         views::all_read(::std::forward<_Range1>(__rng)), views::all_write(::std::forward<_Range2>(__result)));
 }
 
+namespace __internal
+{
+template <typename _BinaryOperation>
+struct __transform_fn2
```

Comment:
We already have more or less "general" `__transform_functor`  in `utils.h`.
Probably, we would re-use one here, w/o introducing new  `struct __transform_fn2` ?

Reply:
Initially it was implemented by this way here by you in the PR https://github.com/uxlfoundation/oneDPL/pull/202